runtime/crab/src/weights/pallet_bridge_parachains.rs

```diff
@@ -57,22 +61,26 @@ impl<T: frame_system::Config> pallet_bridge_parachains::WeightInfo for WeightInf
 	/// Proof: BridgePolkadotParachain PalletOperatingMode (max_values: Some(1), max_size: Some(1), added: 496, mode: MaxEncodedLen)
 	/// Storage: BridgePolkadotGrandpa ImportedHeaders (r:1 w:0)
 	/// Proof: BridgePolkadotGrandpa ImportedHeaders (max_values: None, max_size: Some(65568), added: 68043, mode: MaxEncodedLen)
-	/// Storage: BridgePolkadotParachain ParasInfo (r:1024 w:1024)
+	/// Storage: BridgePolkadotParachain ParasInfo (r:1023 w:1023)
 	/// Proof: BridgePolkadotParachain ParasInfo (max_values: None, max_size: Some(60), added: 2535, mode: MaxEncodedLen)
-	/// Storage: BridgePolkadotParachain ImportedParaHashes (r:1024 w:1024)
+	/// Storage: BridgePolkadotParachain ImportedParaHashes (r:1023 w:1023)
 	/// Proof: BridgePolkadotParachain ImportedParaHashes (max_values: None, max_size: Some(64), added: 2539, mode: MaxEncodedLen)
-	/// Storage: BridgePolkadotParachain ImportedParaHeads (r:0 w:1024)
+	/// Storage: BridgePolkadotParachain ImportedParaHeads (r:0 w:1023)
 	/// Proof: BridgePolkadotParachain ImportedParaHeads (max_values: None, max_size: Some(1092), added: 3567, mode: MaxEncodedLen)
 	/// The range of component `p` is `[1, 1024]`.
-	fn submit_parachain_heads_with_n_parachains(_p: u32, ) -> Weight {
+	fn submit_parachain_heads_with_n_parachains(p: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `485`
-		//  Estimated: `5268275`
-		// Minimum execution time: 76_756_000 picoseconds.
-		Weight::from_parts(39_465_487_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
-			.saturating_add(T::DbWeight::get().reads(2050))
-			.saturating_add(T::DbWeight::get().writes(3072))
+		//  Estimated: `72499 + p * (5074 ±0)`
+		// Minimum execution time: 23_588_000 picoseconds.
+		Weight::from_parts(24_031_000, 0)
+			.saturating_add(Weight::from_parts(0, 72499))
+			// Standard Error: 23_116
+			.saturating_add(Weight::from_parts(18_957_485, 0).saturating_mul(p.into()))
+			.saturating_add(T::DbWeight::get().reads(2))
+			.saturating_add(T::DbWeight::get().reads((2_u64).saturating_mul(p.into())))
+			.saturating_add(T::DbWeight::get().writes((3_u64).saturating_mul(p.into())))
+			.saturating_add(Weight::from_parts(0, 5074).saturating_mul(p.into()))
```

Comment:
What's the p here?

Reply:
https://github.com/darwinia-network/darwinia-messages-substrate/blob/main/modules/parachains/src/weights_ext.rs#L52

It refers to `parachains_count`, it's normally a number less than 10.